### ISSUE TITLE
crash jitting startup methods on Linux x64

### ISSUE THREAD

**Issue by AndyAyersMS** [dotnet/llilc#869]
When trying to run any program on Linux x64 with LLILC, we cause a crash on entry to `System.Buffer.Memmove`. The crash happens immediately on entry in the prolog, trying to save registers onto the stack.
```
   0x7fffe8ffb4c0: subq   $0x28, %rsp
   0x7fffe8ffb4c4: movq   %rdi, 0x20(%rsp)   <--- crash here
   0x7fffe8ffb4c9: movq   %rsi, 0x18(%rsp)
```
For some reason, on entry to this method, `rsp` has a bogus value. The value I see is
```
0x00007ffffbffefd0
```
while reasonable stack values (eg `rbp`) look more like
```
0x00007fffffffd380
```
This method has just been jitted, so the immediate caller is something in the prestub worker chain. Up the stack a bit the managed caller is `System.Buffer.Memcpy`.

Note `Memmove` is the first method jitted that has an `.rodata` contribution, so one guess is that something in the relocation processing for this goes awry and corrupts memory or similar.

**Comment by AndyAyersMS** [comment]
Relocations in the `.rodata` look like this (per LLVM's `-debug -debugonly=dyld`)
```
                RelType: 1 Addend: 76 TargetName:
                This is section symbol
                SectionID: 1 Offset: 0
                RelType: 1 Addend: 633 TargetName:
                This is section symbol
                SectionID: 1 Offset: 8
                RelType: 1 Addend: 666 TargetName:
                This is section symbol
```
so (per #865) we see nonzero Addends, and perhaps we're getting the addend values wrong since this is really a `.rela.rodata`?

**Comment by AndyAyersMS** [comment]
Looks like it is the missing addend that is the culprit. I have some provisional code to handle addends differently for ELF that gets us past `Memmove`.

Also somewhat oddly the section names now no longer have the `.rela` prefix. Not sure what changed.

**Comment by AndyAyersMS** [comment]
With the two recent fixes (#867, #871) basic code-gen bringup tests now pass once again. So, closing.
